lib/vssh/libssh2.c

```diff
@@ -3167,10 +3167,13 @@ static CURLcode ssh_connect(struct Curl_easy *data, bool *done)
 
   sshc = &conn->proto.sshc;
 
-#ifdef CURL_LIBSSH2_DEBUG
```

Comment:
IMO this would better be kept under the DEBUG flag.
Username is sensitive information and should not be
dumped into the log.

Reply:
I don't intend to question your opinion on this, but I'm wondering if I missed that there is precedent around this. (To be fair, I did not try to find documentation about what is considered sensitive and not to be logged.)

* * *

My quick and biased first thought seems to indicate it's at least not sensitive for HTTP.

```
~ $ curl -v 'http://obviouslyusername:obviouslypassword@example.com/' 2>&1 | grep obviously
* Server auth using Basic with user 'obviouslyusername'

~ $ curl -v --user obviouslyusername:obviouslypassword 'http://@example.com/' 2>&1 | grep obviously
* Server auth using Basic with user 'obviouslyusername'
```

Reply:
It's been added in 2003. It was a different environment. That username was sent over the wire unencrypted anyway. Nowadays I think it's not prudent to dump credentials into a log.

Reply:
I think user name is still fine, and since curl already shows the user name in the log for other protocols it seems inconsistent to deny it here. It is useful for debugging.